darts/timeseries.py

```diff
@@ -3,117 +3,554 @@
+            in missing values if some dates are missing and `fill_missing_dates` is set to `True`.
+
+        Returns
+        -------
+        TimeSeries
+            A univariate or multivariate deterministic TimeSeries constructed from the inputs.
+        """
+
+        # optionally fill missing dates; do it only when there is a DatetimeIndex (and not a RangeIndex)
+        if fill_missing_dates and isinstance(xa.get_index(xa.dims[0]), pd.DatetimeIndex):
+            sorted_xa = xa.sortby(xa.dims[0])
+            time_index = sorted_xa.get_index(xa.dims[0])
+
+            if not freq:
+                samples_size = 3
+                observed_frequencies = [
+                    time_index[x:x + samples_size].inferred_freq
+                    for x
+                    in range(len(time_index) - samples_size + 1)]
+
+                observed_frequencies = set(filter(None.__ne__, observed_frequencies))
+
+                raise_if_not(
+                    len(observed_frequencies) == 1,
+                    "Could not infer explicit frequency. Observed frequencies: "
+                    + ('none' if len(observed_frequencies) == 0 else str(observed_frequencies))
+                    + ". Is Series too short (n=2)?",
+                    logger)
+
+                freq = observed_frequencies.pop()
+
+            # TODO: test this
+            # TODO: if provided freq doesn't match the freq in index either raise an error or correct
+            xa_ = sorted_xa.resample({xa.dims[0]: freq}).asfreq()
+        else:
+            xa_ = xa
+
+        # clean components (columns) names if needed (if names are not unique, or not strings)
+        components = xa_.get_index(DIMS[1])
+        if len(set(components)) != len(components) or any([not isinstance(s, str) for s in components]):
+
+            # TODO: test this
+            def _clean_component_list(columns) -> List[str]:
+                # return a list of string containing column names
+                # make each column name unique in case some columns have the same names
+                clist = columns.to_list()
+
+                # convert everything to string if needed
+                for i, column in enumerate(clist):
+                    if not isinstance(column, str):
+                        clist[i] = str(column)
+
+                has_duplicate = len(set(clist)) != len(clist)
+                while has_duplicate:
+                    # we may have to loop several times (e.g. we could have columns ["0", "0_1", "0"] and not
+                    # noticing when renaming the last "0" into "0_1" that "0_1" already exists...)
+                    name_to_occurence = defaultdict(int)
+                    for i, column in enumerate(clist):
+                        name_to_occurence[clist[i]] += 1
+
+                        if name_to_occurence[clist[i]] > 1:
+                            clist[i] = clist[i] + '_{}'.format(name_to_occurence[clist[i]]-1)
```

Comment:
Exactly but it would be "values" and "values_1". Not obvious to which is better (rename both or keep original name), so I'd say let's leave it like that for now.